core/api/src/app/accounts/create-contact.ts

```diff
@@ -0,0 +1,71 @@
+import { CouldNotFindContactFromAccountIdError } from "@/domain/errors"
+import { ContactType } from "@/domain/contacts"
+
+import { ContactsRepository } from "@/services/mongoose"
+
+export const createContact = async ({
+  accountId,
+  handle,
+  displayName,
+  type,
+}: {
+  accountId: AccountId
+  handle: string
+  type: ContactType
+  displayName: string
+}): Promise<Contact | ApplicationError> => {
+  const contactsRepo = ContactsRepository()
+
+  const existing = await contactsRepo.findByHandle({ accountId, handle })
+  if (existing instanceof CouldNotFindContactFromAccountIdError) {
+    return contactsRepo.persistNew({
+      accountId,
+      handle,
+      type,
+      displayName,
+      transactionsCount: 1,
+    })
+  }
+
+  if (existing instanceof Error) return existing
+
+  return contactsRepo.update({
+    ...existing,
+    displayName,
+    transactionsCount: existing.transactionsCount + 1,
```

Comment:
this is not true for mutation, we need a way to disable this, specially for non intraledger contacts